gym/lib/gym/runner.rb

```diff
@@ -72,6 +73,11 @@ def clear_old_files
       end
     end
 
+    def fix_archive
+      return unless Gym.config[:use_legacy_build_api]
+      Gym::XcodebuildFixes.generic_archive_fix
```

Comment:
I'd also really like this to be a feature flag that the user has to enable. Or is there a way to detect if we actually need to apply the fix?